### ISSUE TITLE
Basename of the Unfolded objects

### ISSUE THREAD

**Issue by leoheck** [shaise/FreeCAD_SheetMetal#350]
Hi @shaise , I would like to suggest to use the base Body name (actually Label) as the basename of the unfolded objects.

For instance, the object name is `Bottom`
Then if I unfold it, it could generate the `Bottom_Unfolded` and the `Bottom_Unfolded_Sketch`

![image](https://github.com/shaise/FreeCAD_SheetMetal/assets/1277920/6f2dfa35-840f-49f5-975c-a7990c65c1ec)

**Comment by shaise** [comment]
Sounds like a good idea.
Added to my to do list

**Comment by leoheck** [comment]
Nice, thanks. I tried to improve this here https://github.com/shaise/FreeCAD_SheetMetal/pull/352. What do you think?

**Comment by leoheck** [comment]
The proposal works as you can see in the screenshot below.

I just noticed an interesting thing.
I have enabled the Freecad option to allow object with the same Label.
So no more `001, 002, 003..` at the end of repeated objects.

However when I repeat the unfolding of the same object.
It creates this new Part ending with `001` and a new Sketch with the right name as in my configs. I am not sure if this is an issue with the Sheetmetal or if this is an issue with my latest 0.22-dev build.

What do you think?

![image](https://github.com/shaise/FreeCAD_SheetMetal/assets/1277920/05c24e88-d347-411c-a26c-1cd1363ca3a6)

**Comment by leoheck** [comment]
Ah nevermind, this looks like a bug on the latest dev.

**Comment by shaise** [comment]
of object names are not unique anymore how can you tell them apart?

**Comment by leoheck** [comment]
hm did not know, I got the first parent from the Parents list and on my test, it worked. maybe I got lucky.

**Comment by leoheck** [comment]
Actually all partents are the same, maybe...

This example here:
![image](https://github.com/shaise/FreeCAD_SheetMetal/assets/1277920/e7ab5f9a-d7a2-4bae-9b7a-d01b3a11e795)


```python
Gui.Selection.addSelection('Unnamed','Body001','Bend001.')
>>> ### Begin command Std_SendToPythonConsole
>>> # try:
>>> #     del(doc,lnk,obj,shp,sub,subs)
>>> # except Exception:
>>> #     pass
>>> # 
>>> # doc = App.getDocument("Unnamed")
>>> # obj = doc.getObject("Bend001")
>>> # shp = obj.Shape
>>> ### End command Std_SendToPythonConsole
>>> obj.Label
'Bend001'
>>> obj.Parents
[(<body object>, 'Bend001.'), (<body object>, 'Bend001.')]
>>> obj.Parents[0][1]
'Bend001.'
>>> obj.Parents[0][0].Name
'Body001'
>>> obj.Parents[0][0].Label # <======== Label of the First parent
'my_Body'
>>> obj.Parents[1][0].Label # <======== Label of the Second parent
'my_Body'
>>> 
```

**Comment by shaise** [comment]
I dont think it has to do with object parents. It should be done based on the selected object before the fold.

**Comment by leoheck** [comment]
it can happen too.

**Comment by leoheck** [comment]
the feature is inside a body, the selected body is the same.

**Comment by leoheck** [comment]
can I select multiple bodies to unfold at the same time?

**Comment by shaise** [comment]
no

**Comment by leoheck** [comment]
If I select a face to unfold, then there is no selected body.
![image](https://github.com/shaise/FreeCAD_SheetMetal/assets/1277920/19330f61-a82b-497c-ac5a-9b7360ab2009)

What is selected is a face. Which has a parent body, which is the my_Body in this case

**Comment by leoheck** [comment]
The way I implemented works fine.
